zgw/verzoek/src/test/kotlin/com/ritense/verzoek/VerzoekPluginEventListenerIntTest.kt

```diff
@@ -0,0 +1,364 @@
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.ritense.BaseIntegrationTest
+import com.ritense.document.service.DocumentDefinitionService
+import com.ritense.document.service.DocumentService
+import com.ritense.document.service.result.DeployDocumentDefinitionResult
+import com.ritense.notificatiesapi.event.NotificatiesApiNotificationReceivedEvent
+import com.ritense.objectenapi.ObjectenApiPlugin
+import com.ritense.objectenapi.client.ObjectRecord
+import com.ritense.objectenapi.client.ObjectWrapper
+import com.ritense.objectmanagement.domain.ObjectManagement
+import com.ritense.objectmanagement.service.ObjectManagementService
+import com.ritense.openzaak.domain.request.CreateZaakTypeLinkRequest
+import com.ritense.openzaak.service.ZaakTypeLinkService
+import com.ritense.plugin.domain.PluginConfiguration
+import com.ritense.plugin.domain.PluginConfigurationId
+import com.ritense.plugin.service.PluginService
+import com.ritense.processdocument.service.ProcessDocumentService
+import java.net.URI
+import java.time.LocalDate
+import java.util.*
+import javax.transaction.Transactional
+import org.camunda.bpm.engine.RuntimeService
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doCallRealMethod
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.spy
+import org.mockito.kotlin.whenever
+import org.springframework.beans.factory.annotation.Autowired
+
+@Transactional
+internal class VerzoekPluginEventListenerIntTest : BaseIntegrationTest() {
+
+    private val bsn: String = "999999999"
+    private val initiatoRolType: String = "https://example.com/my-role-type"
+    private val rsin: String = "637549971"
+    private val objectType: String = "anObjectType"
+    private val zaakTypeUrl: String = "http://example.gov"
+    private val verzoekObjectType = "objection"
+
+    lateinit var documentDefinition: DeployDocumentDefinitionResult
+    lateinit var pluginServiceSpy: PluginService
+    lateinit var notificatiesApiPluginConfiguration: PluginConfiguration
+    lateinit var verzoekPluginEventListener: VerzoekPluginEventListener
+    lateinit var objectManagement: ObjectManagement
+
+
+    @Autowired
+    lateinit var processDocumentService: ProcessDocumentService
+
+    @Autowired
+    lateinit var objectManagementService: ObjectManagementService
+
+    @Autowired
+    lateinit var documentService: DocumentService
+
+    @Autowired
+    lateinit var pluginService: PluginService
+
+    @Autowired
+    lateinit var documentDefinitionService: DocumentDefinitionService
+
+    @Autowired
+    lateinit var zaakTypeLinkService: ZaakTypeLinkService
+
+    @Autowired
+    lateinit var processService: RuntimeService
+
+    @BeforeEach
+    fun init() {
+        pluginServiceSpy = spy(pluginService)
```

Comment:
Maybe the pluginservice can be defined as a spybean in the BaseIntegrationTest. That way, the spring context is defined correctly in 1 go. Can save a bit of time when running integration tests.